packages/components/Menu/src/Menu/useMenu.ts

```diff
@@ -61,10 +60,12 @@ const useMenuOpenState = (
 
   const state = isControlled ? open : openInternal;
 
+  const lastCloseTimestampRef = React.useRef<number>(-1);
+
   const setOpen = React.useCallback(
     (e: InteractionEvent, isOpen: boolean, bubble?: boolean) => {
       const openPrev = state;
-      if (!isControlled && (!isOpen || lastCloseTimestamp + delayOpen <= Date.now())) {
+      if (!isControlled && (!isOpen || lastCloseTimestampRef.current + delayOpen <= Date.now())) {
```

Comment:
An aside, you'd probably want to update `useMenu.android` too

Reply:
Fixed!